influxdb/src/main/scala/akka/stream/alpakka/influxdb/impl/InfluxDBFlowStage.scala

```diff
@@ -0,0 +1,131 @@
+    clazz match {
+      case Some(c) => new InfluxDBMapperRecordLogic
+      case None => new InfluxDBRecordLogic
+    }
+
+  sealed abstract class InfluxDBLogic extends GraphStageLogic(shape) with InHandler with OutHandler {
+
+    protected def write(messages: immutable.Seq[InfluxDBWriteMessage[T, C]]): Unit
+
+    setHandlers(in, out, this)
+
+    override def onPull(): Unit = if (!isClosed(in) && !hasBeenPulled(in)) pull(in)
+
+    override def onPush(): Unit = {
+      val messages = grab(in)
+      if (messages.nonEmpty) {
+
+        influxDB.enableBatch(BatchOptions.DEFAULTS)
+        write(messages)
+        val writtenMessages = messages.map(m => new InfluxDBWriteResult(m, None))
+        influxDB.close()
+        push(out, writtenMessages)
+      }
+
+      tryPull(in)
+    }
+  }
+
+  final class InfluxDBRecordLogic extends InfluxDBLogic {
+
+    override protected def write(messages: immutable.Seq[InfluxDBWriteMessage[T, C]]): Unit =
+      messages
+        .filter {
+          case InfluxDBWriteMessage(_: Point, _, _, _) => {
+            true
+          }
+          case InfluxDBWriteMessage(_: AnyRef, _, _, _) => {
+            failStage(new RuntimeException(s"unexpected type Point required"))
+            false
+          }
+        }
+        .groupBy(im => (im.databaseName, im.retentionPolicy))
+        .map(wm => toBatchPoints(wm._1._1, wm._1._2, wm._2))
+        .foreach(influxDB.write)
+
+  }
+
+  final class InfluxDBMapperRecordLogic extends InfluxDBLogic {
+
+    private val mapperHelper: InfluxDBResultMapperHelper = new InfluxDBResultMapperHelper
+
+    override protected def write(messages: immutable.Seq[InfluxDBWriteMessage[T, C]]): Unit =
+      messages
+        .groupBy(groupByDbRp)
+        .map(convertToBatchPoints)
+        .foreach(influxDB.write)
+
+    def groupByDbRp(im: InfluxDBWriteMessage[T, C]) =
+      (
+        im.databaseName match {
+          case Some(databaseName) => Some(databaseName)
+          case None => Some(mapperHelper.databaseName(im.point))
+        },
+        im.retentionPolicy match {
+          case Some(databaseName) => Some(databaseName)
+          case None => Some(mapperHelper.retentionPolicy(im.point))
+        }
+      )
+
+    def convertToBatchPoints(wm: ((Some[String], Some[String]), immutable.Seq[InfluxDBWriteMessage[T, C]])) =
+      toBatchPoints(wm._1._1,
+                    wm._1._2,
+                    wm._2.map(im => im.withPoint(mapperHelper.convertModelToPoint(im.point).asInstanceOf[T])))
+  }
+
+  private def toBatchPoints(databaseName: Option[String],
+                            retentionPolicy: Option[String],
+                            seq: Seq[InfluxDBWriteMessage[T, C]]) = {
+
+    val builder = databaseName match {
+      case Some(databaseName) => BatchPoints.database(databaseName)
+      case None => BatchPoints.builder()
+    }
+
+    if (retentionPolicy.isDefined) builder.retentionPolicy(retentionPolicy.get)
```

Comment:
```suggestion
    retentionPolicy.foreach(p => builder.retentionPolicy(p))
```

Reply:
Taking the suggested change one step further:
```
retentionPolicy.foreach(builder.retentionPolicy)
```